ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -42,6 +48,24 @@ export default class PermissionConnect extends Component {
     originName: this.props.originName,
   }
 
+  componentDidUpdate (prevProps) {
+    const { permissionsRequest, domains } = this.props
+    const { originName } = this.state
+
+    if (permissionsRequest === null && prevProps.permissionsRequest && prevProps.permissionsRequest.permissions) {
```

Comment:
I think we should be checking to see whether this permission request in particular is still being passed in or not. Each permission request should have a unique id in `.metadata.id` I believe.

It's especially important that we get this right, because we're caching the origin in the component but not the permission request. We don't want to get into a situation where we're displaying the origin for a previous request on a different permission request 😬

Reply:
1b68fb882